@types/express/index.d.ts

```diff
@@ -1,14 +1,35 @@
-declare namespace Express {
-  import { Client } from "openid-client";
+import { Client } from "openid-client";
+import { User } from "../../src/types";
+import { QueryParameters } from "../../src/app.constants";
+
+declare module "express-session" {
+  interface Session {
+    timestamp?: number;
+    nonce?: string;
+    state?: string;
+    currentURL?: string;
+    user?: User;
+    user_id?: string;
+    referenceCode?: string;
+    queryParameters: QueryParameters;
```

Comment:
Why is this the only optional parameter in the session?

Reply:
Actually, all are optional but queryParameters. It has been updated now.